### ISSUE TITLE
Anouncement "Boarding completed" when starting deboarding 

### ISSUE THREAD

**Issue by MStallen** [flybywiresim/aircraft#8062]
### Aircraft Version

Development

### Build info

```json
{
    "built": "2023-06-03T15:16:51+00:00",
    "ref": "refs/heads/master",
    "sha": "c24afcf85394a984abb84aa3f36d75e1ed3b8974",
    "actor": "Benjozork",
    "event_name": "manual",
    "pretty_release_name": "master:c24afcf8",
    "version": "v0.10.0-dev.c24afcf"
}
```

### Describe the bug

Before starting A/C start boarding and after a while you will receive the message "Boarding completed". After flight start with deboarding, but receiving again the "Boarding completed" message. And deboarding start only after selecting it twice.

### Expected behavior

No "Boarding completed" message after selecting for deboarding and deboarding starts after first time selection of deboarding on payloadpage

### Steps to reproduce

See description (using simbrief for number of passengers)

**Comment by tracernz** [comment]
Toolbar Pushback addon?

**Comment by GeheimagentNr1** [comment]
@tracernz I reported tla similar issue couple days ago #8056 an in this case it was a livery from flightsim.to. So it is not necessarily the Toolbar Pushback Addon.

**Comment by MattPaxtonBel** [comment]
Same here, no toolbar pushback Addon installed.

**Comment by tracernz** [comment]
> @tracernz I reported tla similar issue couple days ago #8056 an in this case it was a livery from flightsim.to. So it is not necessarily the Toolbar Pushback Addon.

Please tell us which livery it was.

**Comment by MStallen** [comment]
No Toolbar Pushback Addon. Livery : KLM from IFR Liveries flightsim.to

**Comment by edgarcnp** [comment]
I've got the same behaviour when using Lufthansa D-AIJD Livery from flightsim.to, removing it solves the problem. (I don't mind using FBW's livery)

**Comment by GeheimagentNr1** [comment]
> > @tracernz I reported tla similar issue couple days ago #8056 an in this case it was a livery from flightsim.to. So it is not necessarily the Toolbar Pushback Addon.
> 
> Please tell us which livery it was.

I had the issues with:

```json
{
   "built": "2023-05-23T08:10:56+00:00",
   "ref": "refs/heads/experimental",
   "sha": "7ac70bf",
   "actor": "Saschl",
   "event_name": "manual",
   "pretty_release_name": "experimental:7ac70bfd",
   "version": "v0.10.0-exp.7ac70bf"
}
```

But I just tested it with the newest experimental version, but could not reproduce the issues:

```json5
{  
  "built": "2023-06-06T08:54:20+00:00",  
  "ref": "refs/heads/experimental",  
  "sha": "0cb783f",  
  "actor": "Saschl",  
  "event\_name": "manual",  
  "pretty\_release\_name": "experimental:0cb783fe",  
  "version": "v0.10.0-exp.0cb783f"  
}
```

I had this issues with the Liveries:  
https://flightsim.to/file/29262/lufthansa-airbus-a320-neo-d-aino-rastatt-8k  
https://flightsim.to/file/12960/a32nx-flybywire-and-all-mod-airbus-a320neo-lufthansa-d-aija-in-8k-official-realistic   
https://flightsim.to/file/22878/fbw-a32nx-lufthansa-livery-d-aija (Tested with the D-AIJA)

I additionally had the issue, that to the "Boarding complete" audio message, only the cargo and not the PAX were set to 0. #8056  
The double "deboarding" I could only, but not constantly, reproduce with "Instant" Boarding Time.

I just tested, by starting at a gate and loading and unloading the aircraft.  
I have to test, next time I'm flying, if the behaviour is different, after the aircraft was in the air.

**Comment by Nadelbaum** [comment]
Same issue and same behaviour, latest development version, no liveries, no push-back add-ons. After a flight pressing the Deboarding button just gives "Boarding complete". It doesn't seem to change the PAX to 0.

**Comment by GeheimagentNr1** [comment]
https://flightsim.to/file/22878/fbw-a32nx-lufthansa-livery-d-aija (Tested with the D-AIJA)

**Comment by MattPaxtonBel** [comment]
> After a flight I could reproduce the issue like described in #8056 with "Boarding complete" announcement, only the cargo and not the PAX were set to 0, with the livery: https://flightsim.to/file/22878/fbw-a32nx-lufthansa-livery-d-aija (Tested with the D-AIJA)

Did you find in the files and folders of the livery where the problem lays ?

**Comment by edgarcnp** [comment]
Any updates on this issue?

**Comment by GeheimagentNr1** [comment]
I was able to recreate the issue with the default FBW A20NX Livery.

Build Info:
```json
{
    "built": "2023-07-15T17:10:17+00:00",
    "ref": "refs/heads/master",
    "sha": "f261d1508cdff801e4dcaa887a61dac0429297f4",
    "actor": "Benjozork",
    "event_name": "manual",
    "pretty_release_name": "master:f261d150",
    "version": "v0.11.0-dev.f261d15"
}
```

Flight Plan: [Flight_Plan.pdf](https://github.com/flybywiresim/aircraft/files/12062987/Flight_Plan.pdf)
Video: https://github.com/flybywiresim/aircraft/assets/20889593/270550ec-4292-4f6a-a1cb-43daf7d1d694

**Comment by GeheimagentNr1** [comment]
> The de-boarding button only enters "0" into the pax field and starts normal boarding.
> So people experiencing this issue should also test entering "0" in pax and start normal boarding.
> This way we can understand if it is indeed an issue with the de-boarding button or the "0" pax/boarding logic. 
> Report back here please. Thanks

You mean we should first enter 0 into the PAX and then press the deboarding button?

**Comment by frankkopp** [comment]
> > The de-boarding button only enters "0" into the pax field and starts normal boarding.
> > So people experiencing this issue should also test entering "0" in pax and start normal boarding.
> > This way we can understand if it is indeed an issue with the de-boarding button or the "0" pax/boarding logic. 
> > Report back here please. Thanks
> 
> You mean we should first enter 0 into the PAX and then press the deboarding button? 

No, set 0 and press boarding - this is how oit worked before we had the de-boarding button. And as said, this is actually what the de-boarding button does as well. Same logic.

**Comment by Nadelbaum** [comment]
Setting pax to 0 and hitting boarding works, but only after boarding is clicked the second time. The first click just changes the view a bit (brought the Simbrief import button).

**Comment by Nadelbaum** [comment]
I can share a video if needed

**Comment by frankkopp** [comment]
The "Boarding Completed" should sound when boarding or de-boarding is completed - not when started. Here a video I made yesterday - this works as expected.

https://github.com/flybywiresim/aircraft/assets/16833201/f653838b-3184-49ac-906e-5cc7a9f73126

@Nadelbaum if you want to make a video make sure to only have the A32NX in the Community folder and no other tool running (GSX, FSUIPC, Mobiflight, etc.). Best to have a clean install of the latest A32NX version (Dev).
This is super important otherwise the test has no value.

**Comment by Nadelbaum** [comment]
@frankkopp Unfortunately your test is artificial. In reality the Simbrief import button will vanish during a flight. Therefore please check my video, which is from a situation where I have just arrived at a gate after a real flight. Latest Dev version, clean install, no GSX/FSUIPC/Mobiflight etc. involved.


https://github.com/flybywiresim/aircraft/assets/22868316/9cc3c75a-3389-4310-8fb3-1eb6f6bfceb7

**Comment by edgarcnp** [comment]
Hi, I'd like to share some videos too regarding this issue as I just finished a flight and experienced this issue. Also no other addons are in the community folder except the A32NX and Simbridge.

1st Video:
Arrived at the gate and proceed to deboard the passengers by clicking the deboarding button on the payload page in the EFB. After I pressed the button only the cargo drop to 0 while the pax remains the same.
https://github.com/flybywiresim/aircraft/assets/56579802/d5ab32f9-e37b-4687-98aa-212a3587fbcf

2nd Video:
In order to deboard the pax I need to cancel the deboarding process and click the deboarding button for the 2nd time. Then the pax get deboarded.
https://github.com/flybywiresim/aircraft/assets/56579802/8bc97a45-1a39-4ef3-a125-71bda2542bd7

3rd Video:
As per @frankkopp request, I filled up the pax again using the load from simbrief then put 0 in the pax. The deboarding is commenced without a problem.
https://github.com/flybywiresim/aircraft/assets/56579802/8378d575-ba86-4de8-b2ce-e606f1517d0e

Notes:
This bug only happen everytime I press the deboarding button for the **first time**, when I do a round trip the deboarding button works as intended the second time I pressed it. Here's a video proofing that the deboarding button works after the first try:
https://github.com/flybywiresim/aircraft/assets/56579802/e95ee743-342b-4538-98d0-103aba4fa068

Conclusion:
In my opinion, the deboarding logic itself is functioning correctly, there is an issue with the deboarding button's behavior during the first use in a game session. It seems to be bugged or not working properly, which may prevent players or users from successfully initiating the deboarding process when they try to do so for the first time.

**Comment by frankkopp** [comment]
> @frankkopp Unfortunately your test is artificial. In reality the Simbrief import button will vanish during a flight. Therefore please check my video, which is from a situation where I have just arrived at a gate after a real flight. Latest Dev version, clean install, no GSX/FSUIPC/Mobiflight etc. involved.
> 
>  VID_20230728_201214.2.mp4

SimBrief Import appears as long as the current target load is different from the SimBrief target load. Once everything is loaded the button disappears. 

And my test is not artificial - it simply shows that in a normal setting it works as expected. Not sure what happens on your side but I will test a few things.

**Comment by frankkopp** [comment]
> Hi, I'd like to share some videos too regarding this issue as I just finished a flight and experienced this issue. Also no other addons are in the community folder except the A32NX and Simbridge.
> 
> 1st Video: Arrived at the gate and proceed to deboard the passengers by clicking the deboarding button on the payload page in the EFB. After I pressed the button only the cargo drop to 0 while the pax remains the same. [flybywiresim/aircraft/assets/56579802/d5ab32f9-e37b-4687-98aa-212a3587fbcf](https://github.com/flybywiresim/aircraft/assets/56579802/d5ab32f9-e37b-4687-98aa-212a3587fbcf)
> 
> 2nd Video: In order to deboard the pax I need to cancel the deboarding process and click the deboarding button for the 2nd time. Then the pax get deboarded. [flybywiresim/aircraft/assets/56579802/8bc97a45-1a39-4ef3-a125-71bda2542bd7](https://github.com/flybywiresim/aircraft/assets/56579802/8bc97a45-1a39-4ef3-a125-71bda2542bd7)
> 
> 3rd Video: As per @frankkopp request, I filled up the pax again using the load from simbrief then put 0 in the pax. The deboarding is commenced without a problem. [flybywiresim/aircraft/assets/56579802/8378d575-ba86-4de8-b2ce-e606f1517d0e](https://github.com/flybywiresim/aircraft/assets/56579802/8378d575-ba86-4de8-b2ce-e606f1517d0e)
> 
> Notes: This bug only happen everytime I press the deboarding button for the **first time**, when I do a round trip the deboarding button works as intended the second time I pressed it. Here's a video proofing that the deboarding button works after the first try: [flybywiresim/aircraft/assets/56579802/e95ee743-342b-4538-98d0-103aba4fa068](https://github.com/flybywiresim/aircraft/assets/56579802/e95ee743-342b-4538-98d0-103aba4fa068)
> 
> Conclusion: In my opinion, the deboarding logic itself is functioning correctly, there is an issue with the deboarding button's behavior during the first use in a game session. It seems to be bugged or not working properly, which may prevent players or users from successfully initiating the deboarding process when they try to do so for the first time.

You videos show that it the problem might already be in the logic and not the deboarding button which is also my suspicion. Will do a few more tests based on the feedback here.

**Comment by frankkopp** [comment]
@Nadelbaum figured out your "issue" - you are in the pax field with the pax field still active and you try to click directly on the button without leaving the field first. So your click on the button is not really registered and only counts as leaving the entry field. The second click then presses the button. 

See this video - first I click once outside the entry field then the button - works. 
Second try I directly try to click the button and it only registers that I left the entry field. The second click then presses the button.

Although it is possible this can be improved this is currently working as intended. 

https://github.com/flybywiresim/aircraft/assets/16833201/64d0f805-8482-495a-ab0e-6adcabb4c98a

**Comment by frankkopp** [comment]
@edgarcnp I wasn't able to reproduce this (Video 1) but I made a small fix in this PR https://github.com/flybywiresim/aircraft/pull/8137 for a theoretical code issue. Theoretical as I can't really see how this could cause an issue in reality but one never knows. 

Are you able to test a PR? As I said I can't reproduce it on my system and as my fix targets a code timing issue it might behave differently on different systems.

**Comment by Nadelbaum** [comment]
@frankkopp Here's another video where I just, after landing and taxing back to gate, just click on the deboarding button. As you can see from the video I have to do it two times. Dev version updated today before the flight.


https://github.com/flybywiresim/aircraft/assets/22868316/99b08fa1-3fb7-4982-8e12-1d9a2f7b7425

**Comment by edgarcnp** [comment]
> @edgarcnp I wasn't able to reproduce this (Video 1) but I made a small fix in this PR https://github.com/flybywiresim/aircraft/pull/8137 for a theoretical code issue. Theoretical as I can't really see how this could cause an issue in reality but one never knows. 
> 
> Are you able to test a PR? As I said I can't reproduce it on my system and as my fix targets a code timing issue it might behave differently on different systems. 

Yeah. I'll test the #8137 PR for this issue and will let you know ASAP.

**Comment by edgarcnp** [comment]
Hi @frankkopp after I did a test flight with A32NX from the #8137 unfortunately the issue is still there, here's a video of the deboarding process:
https://github.com/flybywiresim/aircraft/assets/56579802/61774653-e150-43ca-9097-d18a05313502

Here's my community folder:
![image](https://github.com/flybywiresim/aircraft/assets/56579802/e6bea901-c7a3-4a56-a99e-c36a8508bd7a)

And here's the build info:
```
{
    "built": "2023-07-28T12:37:00+00:00",
    "ref": "refs/pull/8137/merge",
    "sha": "46977dc903cc3613ea9e6a7bf5645d33ea358daf",
    "actor": "frankkopp",
    "event_name": "manual",
    "pretty_release_name": "HEAD:46977dc9",
    "version": "v0.11.0-8137.46977dc"
}
```
If you need more testing please let me know, I'm happy to help.

**Comment by frankkopp** [comment]
@edgarcnp thanks for testing - much appreciated. 

I already suspected that my little fix wouldn't help. 

Can you confirm this only happens after a flight?

Anyway, not sure what is causing this and I probably won't spend more time debugging this as boarding and loading is currently completely re-written in Rust. The UI will have to be adapted then anyway so it is not worth investing any more time into this. Especially as there is such an easy workaround.

**Comment by edgarcnp** [comment]
@frankkopp 
> Can you confirm this only happens after a flight?

Yes, this bug only happens after a flight. I did test it about a week ago when I deboarded the pax before leaving the gate, and the deboarding button is functioning as normal.

> Anyway, not sure what is causing this and I probably won't spend more time debugging this as boarding and loading is currently completely re-written in Rust. The UI will have to be adapted then anyway so it is not worth investing any more time into this. Especially as there is such an easy workaround.

I totally understand that, guess we'll wait till the Rust rewrite is done. Thanks for your time.

**Comment by larsll** [comment]
Is it expected that the full boarding completed announcement should come after deboarding? Kind of annoying when I am about to make all go cold and dark…

**Comment by 2hwk** [comment]
Check if fixed after https://github.com/flybywiresim/aircraft/pull/8143 (latest dev merge)

**Comment by edgarcnp** [comment]
> Check if fixed after #8143 (latest dev merge)

Hey, I did a test with one of your commit in that PR yesterday and this issue never happened again.

**Comment by 2hwk** [comment]
> > Check if fixed after #8143 (latest dev merge)
> 
> Hey, I did a test with one of your commit in that PR yesterday and this issue never happened again.

I'll leave it open for a bit until it is confirmed, this has now been merged into the main development branch so you can just go off the normal installer build now, thanks for looking into it.

**Comment by edgarcnp** [comment]
> > > Check if fixed after #8143 (latest dev merge)
> > 
> > 
> > Hey, I did a test with one of your commit in that PR yesterday and this issue never happened again.
> 
> I'll leave it open for a bit until it is confirmed, this has now been merged into the main development branch so you can just go off the normal installer build now, thanks for looking into it.

Hello again, I just did a complete turn-around flight with the latest dev version 203cfdd (main branch) and can confirm that the issue where the pax didn't change to 0 when the deboarding process is initiated doesn't exist anymore since #8143 is merged. You can close this issue as solved. Thank you dev team.

**Comment by edgarcnp** [comment]
Here's a video of my arrival using the latest dev 12c8e35 (I don't have time to edit it so you can just skip it to the deboarding process)
Uploading 2023-09-05 05-38-57.mp4…

**Comment by 2hwk** [comment]
Boarding Completed sound issue to be fully addressed by https://github.com/flybywiresim/aircraft/pull/8197
